modules/core/include/opencv2/core/hal/intrin_vsx.hpp

```diff
@@ -1524,6 +1538,82 @@ OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_uint32x4, vec_uint4)
 OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_int32x4, vec_int4)
 OPENCV_HAL_IMPL_VSX_TRANSPOSE4x4(v_float32x4, vec_float4)
 
+template<int i>
+inline v_int8x16 v_broadcast_element(v_int8x16 v)
```

Comment:
I know its too late, but GCC will not able to optimize such permute like this one into `vsplt[b,h,w]` instructions, intrinsic `vec_splat()` must be used instead. in case of use `vec_splat()` with double word vector, the compiler will use the instruction `xxpermdi` A.K.A `vec_permi()` which is far way better than `vperm`.

Reply:
Feel free to propose patch for this.

Reply:
@alalek I think I need to create another PR for this right?

Reply:
oh never mind, @seiko2plus already fixed it. Thanks.